### DISCUSSION TITLE
How do we lint the translate key

### DISCUSSION THREAD

**Discussion by cakasuma** [fnando/i18n#34] [Q&A] [answered]
currently everytime we want to translate, the following syntax used to access the translation store in json

-json
```
{
  "mykey": {
    "message": "hello world"
  }
}
```

-usage
```
i18n.t('mykey.message')
```

this come as a problem when i try to access the translation as the following
```
i18n.t('mykey.xxxx')
```
since the key is not available, but my editor does not highlight me anything that xxxx is not available in my json

is there a way to let my editor know maybe a eslint plugin that the key is missing or do we have existing build command used to check if the key accessed are correct?

**Comment by fnando** [comment, marked as answer]
Hi @cakasuma. I recommend you check https://github.com/fnando/i18n-js out. It includes two commands to help detecting missing translations.

- `i18n lint:translations` check if you have missing translations compared to the default locale
- `i18n lint:scripts` check if your script files (.js, .ts, .jsx, .tsx) reference missing keys (like what you asked for).

**Reply by cakasuma** [reply]
how do i add this to my npm project? (not a ruby project)

**Reply by cakasuma** [reply]
cc: @fnando

**Reply by fnando** [reply]
I don't intend to port the ruby CLI to JavaScript, unfortunately. You can either introduce ruby as a development dependency of your project or create and maintain an equivalent CLI in JavaScript (I won't accept this implementation to the mainstream project though, given that I don't want to maintain it).
